std/conv.d

```diff
@@ -543,6 +543,13 @@ template to(T)
     }
 }
 
+private T toImpl(T, S)(S value)
+if (isInputRange!S && isInfinite!S && isExactSomeString!T)
+{
+    auto _ = value;
+    static assert(0, "Cannot convert infinite range to string. Use `.take(n)` or `.array` to make it finite.");
```

Comment:
`array` requires a finite range (since like `to`, it's converting the entire range). If you want another suggestion, then `takeExactly` would be appropriate.

Also, you should say `std.range.take`, not `.take`. Having a period in front of the symbol indicates that it's supposed to be taken from the outer scope, which makes no sense here.

Reply:
Also, there should be a test that tests that using `to` on an infinite range doesn't compile. Arguably, there already should have been one, but if you're specifically adding an overload like this, then it needs one even more.